specification/security/resource-manager/Microsoft.Security/preview/2023-05-01-preview/customRecommedations.json

```diff
@@ -0,0 +1,384 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft Defender for Cloud",
+    "description": "API spec for Microsoft.Security (Microsoft Defender for Cloud) resource provider",
+    "version": "2023-05-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/{scope}/providers/Microsoft.Security/customRecommendations": {
```

Comment:
>scope [](http://example.com/codeflow?start=7&length=5)

When the scope is Security Connector, the resource id is becoming invalid here, due to /providers/Microsoft.Security segment appearing twice. I would suggest to list resource type separately for all scopes to ensure all the explicitly valid, instead of using {scope}.

Reply:
Not sure why this is a problem, a proxy resource can be extension to another resource. In this case the customRecommendations is an extension to the security connector.
I see live examples in the manifest for this behavior in governancerules.

Reply:
I believe this is the correct format for extension resources; however, what scopes are actually supported? Is it only securityConnectors? We are trying to tighten up specs by enumerating paths explicitly unless there are a lot of valid scopes. Is it possible to enumerate paths for all supported scopes, or are there too many?

Reply:
As I stated in the description of the scope parameters:
Valid scopes are: management group (format: 'providers/Microsoft.Management/managementGroups/{managementGroup}'), subscription (format: 'subscriptions/{subscriptionId}'), or security connector (format: 'subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/securityConnectors/{securityConnectorName})'.

for the he standard assignments type there can be a lot more because we also support resource ids, I want the specs for all types to look the same.